### ISSUE TITLE
BuildToolCLI strips META-INF from jars

### ISSUE THREAD

**Issue by nikita-island** [msintuneappsdk/ms-intune-app-sdk-android#178]
## Questions to Ask Before Submission

1. Does your app compile and launch successfully without the Intune App SDK? **Yes**
2. Have you checked the [Microsoft Intune App SDK for Android] repository for similar issues? **Yes**
3. Are you using the latest version of the [SDK and Plugin]? **Yes**

## Summary

When processing `.jar`s with `BuildToolCLI`, the tool removes `META-INF` from the output `.jar`.
This creates various problems. For example, some `.jar`s contain embedded proguard config `META-INF/com.android.tools/r8/*`, and stripping them causes compilation errors.

Example of a public `.jar` where proguard config gets stripped: [org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.6.4](https://mvnrepository.com/artifact/org.jetbrains.kotlinx/kotlinx-coroutines-core-jvm/1.6.4)

Note that when the `--input` is a directory rather than a `.jar`, `BuildToolCLI` copies `META-INF` correctly.

## Details

- **Intune Android App SDK Version:** 9.7.0
- **Gradle Version:** N/A, this is standalone `BuildToolCLI`
- **AGP Version:** N/A, this is standalone `BuildToolCLI`
- **Operating System (Mac, Linux, Windows) and Version:** Linux (Fedora 37)
- **Android Studio Version:** N/A, this is standalone `BuildToolCLI`

### Android Studio / Gradle Logs

N/A, no gradle or Android Studio. This is standalone `BuildToolCLI`.
It doesn't print any errors, rewrite completes successfully. Downstream dex-ing fails with the rewritten `.jar`s.

### Gradle Build Plugin Report Logs

N/A, no gradle or Android Studio. This is standalone `BuildToolCLI`.
It doesn't print any errors, rewrite completes successfully. Downstream dex-ing fails with the rewritten `.jar`s.

**Comment by boyzhang-msft** [comment]
Hi @nikita-island , thanks for reporting this issue. We have created a bug (ADO# 25068431) internally to track. I'd like to ask a few clarification questions please:
- Is your app a native Android app? Or is it built with some other frameworks like Xamarin?
- Does it work when you put the jar file in a directory, and provide the directory as --input?

**Comment by nikita-island** [comment]
@boyzhang-msft:

> * Is your app a native Android app? Or is it built with some other frameworks like Xamarin?

Yes, it's a native app (with a custom build system that doesn't use Gradle)

> * Does it work when you put the jar file in a directory, and provide the directory as --input?

Yes, as I mentioned in the issue, using an extracted .jar works correctly.

**Comment by nikita-island** [comment]
@boyzhang-msft Hi, any updates on this issue?

**Comment by boyzhang-msft** [comment]
@nikita-island , we do not have an ETA yet. I'll keep you posted here once we made progress. Sorry about the delay!
